programs/manifest/src/state/market.rs

```diff
@@ -0,0 +1,1077 @@
+            // cancelling because of rounding, this counters that. This ensures
+            // that the amount of quote that the maker has credit for when they
+            // cancel/expire is always the maximum amount that could have been
+            // used in matching that order. This atom is not lost, it just is
+            // now sitting in the RestingOrder.
+            if !is_bid {
+                // !is_bid means the maker is a bid, so we need to adjust quote
+                // for rounding.
+                update_balance_by_trader_index(
+                    dynamic,
+                    other_trader_index,
+                    is_bid,
+                    true,
+                    (previous_maker_quote_atoms_allocated
+                        .checked_sub(new_maker_quote_atoms_allocated)?
+                        .checked_sub(quote_atoms_traded)?)
+                    .as_u64(),
+                )?;
+            }
+
+            // Decrease taker
+            update_balance_by_trader_index(
+                dynamic,
+                trader_index,
+                !is_bid,
+                false,
+                if is_bid {
+                    quote_atoms_traded.into()
+                } else {
+                    matched_num_base_atoms.into()
+                },
+            )?;
+            // Increase taker
+            update_balance_by_trader_index(
+                dynamic,
+                trader_index,
+                is_bid,
+                true,
+                if is_bid {
+                    matched_num_base_atoms.into()
+                } else {
+                    quote_atoms_traded.into()
+                },
+            )?;
+
+            // Lookup on every match. This is relatively cheap and fills can
+            // afford the CU of repeated lookups.
+            let trader: Pubkey = get_helper::<RBNode<ClaimedSeat>>(dynamic, trader_index)
+                .get_value()
+                .trader;
+            emit_stack(FillLog {
+                market,
+                maker: get_helper::<RBNode<ClaimedSeat>>(dynamic, other_trader_index)
+                    .get_value()
+                    .trader,
+                taker: trader,
+                base_atoms: matched_num_base_atoms,
+                quote_atoms: quote_atoms_traded,
+                price: matched_price,
+                taker_is_buy: PodBool::from(is_bid),
+                _padding: [0; 15],
+            })?;
+
+            if did_fully_match_resting_order {
+                trace!("FULL MATCH");
+                remove_order_from_tree_and_free(fixed, dynamic, current_order_index, !is_bid)?;
+                remaining_base_atoms = remaining_base_atoms.checked_sub(matched_num_base_atoms)?;
+                current_order_index = next_order_index;
+            } else {
+                trace!(
+                    "PART MATCH -> reduce {} by {matched_num_base_atoms}",
```

Comment:
Partial better?

Reply:
I will soon get rid of all of this logging.